config/lang/check_types.go

```diff
@@ -9,9 +9,21 @@ import (
 
 // TypeCheck implements ast.Visitor for type checking an AST tree.
 // It requires some configuration to look up the type of nodes.
+//
+// It also optionally will not type error and will insert an implicit
+// type conversions for specific types if specified by the Implicit
+// field. Note that this is kind of organizationally weird to put into
+// this structure but we'd rather do that than duplicate the type checking
+// logic multiple times.
 type TypeCheck struct {
 	Scope *Scope
 
+	// Implicit is a map of implicit type conversions that we can do,
+	// and that shouldn't error. The key of the first map is the from type,
+	// the key of the second map is the to type, and the final string
+	// value is the function to call (which must be registered in the Scope).
+	Implicit map[ast.Type]map[ast.Type]string
```

Comment:
Hm, mostly because in terms of the AST tree, there is no way to register the function body into it, since all functions are implemented outside the AST tree (the language doesn't allow function definitions).